slsDetectorServers/mythen3DetectorServer/RegisterDefs.h

```diff
@@ -2,39 +2,166 @@
 #pragma once
 
 /* Definitions for FPGA*/
-#define MEM_MAP_SHIFT 1
 
+#define BASE_CONTROL                    (0x0)
+#define BASE_PATTERN_CONTROL            (0x200 )
+#define BASE_PATTERN_RAM                (0x10000 )
 
+/* Basic detector FPGA registers --------------------------------------------------*/
+/* FPGA Version register */
+#define FPGA_VERSION_REG                (0x04 + BASE_CONTROL)
 
+#define FPGA_COMPILATION_DATE_OFST		(0)
+#define FPGA_COMPILATION_DATE_MSK		(0x00FFFFFF << FPGA_COMPILATION_DATE_OFST)
+#define DETECTOR_TYPE_OFST   			(24)
+#define DETECTOR_TYPE_MSK   			(0x000000FF << DETECTOR_TYPE_OFST)
+
+
+/* Module Control Board Serial Number Register */
+#define MCB_SERIAL_NO_REG               (0x00 + BASE_CONTROL)
+
+/* API Version Register */
+#define API_VERSION_REG                 (0x08 + BASE_CONTROL)
+
+/* Fix pattern register */
+#define FIX_PATT_REG             	    (0x0C + BASE_CONTROL)
+#define FIX_PATT_VAL                    (0xACDC2019)
 
-#ifdef VIRTUAL // until firmware is ready ----------------------------------
 /* Status register */
-#define STATUS_REG            			(0x01 << MEM_MAP_SHIFT)
+#define STATUS_REG                      (0x10 + BASE_CONTROL)
 
-#define RUN_BUSY_OFST					(0)
-#define RUN_BUSY_MSK      				(0x00000001 << RUN_BUSY_OFST)
+/* Look at me register */
+#define LOOK_AT_ME_REG          		(0x14 + BASE_CONTROL)	//Not used in firmware or software, good to play with
 
-/* Set Cycles 64 bit register */
-#define SET_CYCLES_LSB_REG    			(0x02 << MEM_MAP_SHIFT)
-#define SET_CYCLES_MSB_REG    			(0x03 << MEM_MAP_SHIFT)
 
-/* Set Frames 64 bit register */
-#define SET_FRAMES_LSB_REG   			(0x04 << MEM_MAP_SHIFT)
-#define SET_FRAMES_MSB_REG    			(0x05 << MEM_MAP_SHIFT)
+/* Pattern Control FPGA registers --------------------------------------------------*/
+/* Pattern status Register*/
+#define PAT_STATUS_REG                  (0x00 + BASE_PATTERN_CONTROL)
 
-/* Set Period 64 bit register tT = T x 50 ns */
-#define SET_PERIOD_LSB_REG    			(0x06 << MEM_MAP_SHIFT)
-#define SET_PERIOD_MSB_REG    			(0x07 << MEM_MAP_SHIFT)
+/* Delay left 64bit Register */
```

Comment:
Get cycles and get delay, is this really needed? are these registers implemented in the firmware?